vertx-web/src/main/java/io/vertx/ext/web/sstore/impl/SessionImpl.java

```diff
@@ -74,6 +75,18 @@ public String id() {
   }
 
   @Override
+  public Session regenerateId() {
+    if (oldId == null) {
+      // keep track of just the first one since the
+      // regeneration during the remaining lifecycle are ephemeral
+      oldId = id;
+    }
+    id = UUID.randomUUID().toString();
```

Comment:
This probably isn't the safest way to generate a session key, and it doesn't respect the minimum length. It generates 16 bytes of token, but some of that is not random. It initializes SecureRandom using potentially insecure default algorithms, and never re-seeds. It also may block, possibly for a seeming eternity if your underlying platform doesn't have good entropy.